src/docfx/config/ops/OpsConfigLoader.cs

```diff
@@ -151,28 +152,50 @@ private static bool NeedRunLearnValidation(OpsDocsetConfig? docsetConfig)
             && docsetConfig.CustomizedTasks.TryGetValue("docset_postbuild", out var plugins)
             && plugins.Any(plugin => plugin.EndsWith("TripleCrownValidation.ps1", StringComparison.OrdinalIgnoreCase));
 
-        private static JObject GenerateJoinTocMetadata(OpsJoinTocConfig[] configs, string buildSourceFolder)
+        private static (JObject joinTocMetadata, JArray joinTocConfig) GenerateJoinTocMetadataAndConfig(
```

Comment:
consider split into two functions

Reply:
would like to use one loop to generate both